esm-samples/jsapi-oauth/main.js

```diff
@@ -0,0 +1,75 @@
+import "./style.css";
+
+import esriId from "@arcgis/core/identity/IdentityManager";
+import OAuthInfo from "@arcgis/core/identity/OAuthInfo";
+import Portal from "@arcgis/core/portal/Portal";
+
+// UI elements
+const userDetails = document.getElementById("user-display");
+const signInButton = document.getElementById("sign-in-button");
+signInButton.addEventListener("click", signInOrOut);
+
+//Create a new OAuthInfo object.
+const info = new OAuthInfo({
+  // Swap this ID out with an app ID registered in your ArcGIS Organization.
+  appId: "REGISTERED_CLIENT_ID"
+  // Add the portalUrl property if using your own portal.
+  //portalUrl: "https://<host>:<port>/<webadaptor>",
+  // Set the authNamespace property to prevent the user's signed in state
+  // from being shared with other apps on the same domain with the same authNamespace value.
+  // authNamespace: "portal_oauth_inline",
+  // Set popup to true to show the OAuth sign-in page in a separate popup window.
+  //popup: true
+});
+
+// Add the OAuthInfo to the IdentityManager.
+esriId.registerOAuthInfos([info]);
+
+// Call the checkSignIn function to see if the user is already signed in.
+checkSignIn();
+
+// Function to sign in or out of the portal
+function signInOrOut() {
```

Comment:
Will this still work if you convert the classic functions to arrow functions? That's the pattern used in the other samples.

Reply:
It should! I'll update and test it again.